File: fsharp/log-levels/LogLevels.fs
module LogLevels

let message (logLine: string): string = 
    let r = logLine.Split ":"
    r[1].Trim()

let logLevel(logLine: string): string = 
    let r = logLine.Split ":"
    let r0 = r[0].Replace("[", "")
    let r1 = r0.Replace("]", "")
    r1.ToLower()

let reformat(logLine: string): string = 
    let m = message(logLine)
    let l = logLevel(logLine)

    sprintf "%s (%s)" m l

File: java/log-levels/src/main/java/LogLevels.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class LogLevels {
    
    public static String message(String logLine) {
        Pattern pattern = Pattern.compile("^\\[.+\\]:\\s+(.*)$");
        Matcher m = pattern.matcher(logLine);
        return m.find() ? m.group(1).trim() : "";
    }

    public static String logLevel(String logLine) {
        Pattern pattern = Pattern.compile("^\\[(.+)].*");
        Matcher m = pattern.matcher(logLine);
        return m.find() ? m.group(1).toLowerCase() : "";
    }

    public static String reformat(String logLine) {
        return String.format("%s (%s)", message(logLine), logLevel(logLine));
    }
}
